extension/src/heartbeat_agg.rs

```diff
@@ -20,16 +20,19 @@ const BUFFER_SIZE: usize = 1000; // How many values to absorb before consolidati
 pub struct HeartbeatTransState {
     start: i64,
     end: i64,
+    last: i64,
     interval_len: i64,
     buffer: Vec<i64>,
     liveness: Vec<(i64, i64)>, // sorted array of non-overlapping (start_time, end_time)
 }
 
 impl HeartbeatTransState {
     pub fn new(start: i64, end: i64, interval: i64) -> Self {
+        assert!(end - start > interval);
```

Comment:
Can this be triggered by user input?  If so, is the error message comprehensible and can we test it?

Same for another assertion below.

Reply:
I've added a more explicit error message here and to insert that should make it clear to users what went wrong in this case.